test/functional/pdo_sqlsrv/pdo_param_output_select_variant.phpt

```diff
@@ -65,12 +71,15 @@ try {
     dropTable($conn, $tableName);
     unset($conn);
 } catch (Exception $e) {
-    echo $e->getMessage();
+    $error = $e->getMessage();
+    if (!(!isAEConnected() && strpos($error, "Implicit conversion from data type sql_variant to nvarchar(max) is not allowed. Use the CONVERT function to run this query.") !== false) &&
+        !(isAEConnected() && strpos($error, "Invalid Descriptor Index") !== false)) {
+        echo $error;
```

Comment:
Is it possible to simplify the conditions and error checking here? Perhaps use a helper method with the expected message? The same applies to the other test in this PR that uses the very similar logic.